Sources/CoalView/Components/List/CoalCategoryGridView.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  CoalCategoryGridView.swift
+//
+//
+//  Created by M. Rizki Maulana on 30/09/24.
+//
+
+import SwiftUI
+import CoalModel
+
+public enum LayoutDirection {
+  case horizontal
+  case vertical
+}
+
+public struct CoalCategoryGridView: View {
+  let categories: [CategoryModel]
+  var layoutType: LayoutDirection
+  var horizontalRows: Int = 0
+  var verticalRows: Int = 0
+  
+  var columns: [GridItem] {
+    switch layoutType {
+    case .horizontal:
+      return Array(repeating: GridItem(.fixed(100)), count: horizontalRows)
+    case .vertical:
+      return Array(repeating: GridItem(.flexible()), count: verticalRows)
+    }
+  }
+  
+  public init(categories: [CategoryModel], layoutType: LayoutDirection, horizontalRows: Int, verticalRows: Int) {
+    self.categories = categories
+    self.layoutType = layoutType
+    self.horizontalRows = horizontalRows
+    self.verticalRows = verticalRows
+  }
+  
+  public var body: some View {
+    Group {
+      switch layoutType {
+      case .horizontal:
+        ScrollView(.horizontal, showsIndicators: false) {
+          LazyHGrid(rows: columns, spacing: 16) {
+            ForEach(categories) { category in
+              CoalCategoryView(category: category)
+            }
+          }
+          .padding([.leading, .trailing], 20)
+        }
+      case .vertical:
+        ScrollView(.vertical, showsIndicators: false) {
+          LazyVGrid(columns: columns, spacing: 16) {
+            ForEach(categories) { category in
+              CoalCategoryView(category: category)
+            }
+          }
+          .padding([.leading, .trailing], 20)
+        }
+      }
+    }
+  }
+}
```

Comment:
ini kan yg ngebedain .horizontal dan .vertical aja ya. better ScrollView block dipindah ke method terpisah biar tinggal reuse ga perlu di tulis ulang seblock scrollview.

Reply:
Dibuat gini kah mas?

private var horizontalScrollView: some View {
    ScrollView(.horizontal, showsIndicators: false) {
      LazyHGrid(rows: columns, spacing: 16) {
        ForEach(categories) { category in
          CoalCategoryView(category: category)
        }
      }
      .padding([.leading, .trailing], 20)
    }
}
  
private var verticalScrollView: some View {
    ScrollView {
      LazyVGrid(columns: columns, spacing: 16) {
        ForEach(categories) { category in
          CoalCategoryView(category: category)
        }
      }
      .padding([.leading, .trailing], 20)
    }
}
  
private var scrollView: some View {
    Group {
      switch layoutType {
      case .horizontal:
        horizontalScrollView
      case .vertical:
        verticalScrollView
      }
    }
}